mailgun/test_scripts/tlt-2108-handle-missing-attachments-integration-test.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+import hashlib
+import hmac
+import json
+import time
+import uuid
+
+import requests
+from django.conf import settings
+
+try:
+    settings.NO_REPLY_ADDRESS
+except ImportError:
+    # make sure the project root is in sys.path
+    import os, sys
+    sys.path.insert(0, os.path.normpath(
+                           os.path.join(os.path.abspath(__file__), '..', '..', '..')))
+
+def generate_signature_dict():
+    timestamp = str(time.time())
+    token = str(uuid.uuid4())
+    signature = hmac.new(
+        key=settings.LISTSERV_API_KEY,
+        msg='{}{}'.format(timestamp, token),
+        digestmod=hashlib.sha256
+    ).hexdigest()
+    return {
+        'timestamp': timestamp,
+        'token': token,
+        'signature': signature
+    }
+
+# prep the post body
+list_address = 'canvas-6760-2069@mg.dev.tlt.harvard.edu'
+post_body = {
+    'sender': 'integrationtest@example.edu',
+    'recipient': list_address,
+    'from': 'Integration Test <integrationtest@example.edu>',
+    'subject': 'blah',
+    'body-plain': 'blah blah',
+    'attachment-count': 1,
+    'To': list_address,
+}
+post_body.update(generate_signature_dict())
+
+# figure out which server to post to
+env_name = settings.ENV_NAME
+if env_name not in ('dev', 'qa', 'stage'):
+    env_name = 'dev'
+
+# post it
+url = 'https://lti-emailer.%s.tlt.harvard.edu/mailgun/handle_mailing_list_email_route/' % env_name
+resp = requests.post(url, data=post_body)
+if not resp.status_code == 400:
```

Comment:
:) That's a matter of preference, I guess. If QA is running these things, my thought is they'd prefer verbose output. As long as we're all on the same page. If they're going to act like a unix command, maybe the comments about how they're used and what to expect could just be code comments. I don't care that much, but what it means to act unix-y isn't something everyone knows about, or even those who do (like me) don't necessarily assume scripts work like that. I'm usually much better at asking others for documentation than writing it myself, it must be added.